sympy/functions/elementary/piecewise.py

```diff
@@ -1017,43 +1017,41 @@ def _clip(A, B, k):
     oo = S.Infinity
     a, b = B
     c, d = A
-    a = Min(a, b)
     c, d = Min(Max(c, a), b), Min(Max(d, a), b)
```

Comment:
This makes  c  and  d  lie between  a  and  b  (assuming that  a <= b), so  b <= c  means that  c == d == b and  d <= a  means that  a == c  == d, and in those cases the  k -interval has length zero.